src/utils.ts

```diff
@@ -590,7 +591,7 @@ export class MongoDBNamespace {
   static fromString(namespace?: string): MongoDBNamespace {
     if (!namespace) {
       // TODO(NODE-3483): Replace with MongoNamespaceError
-      throw new MongoDriverError(`Cannot parse namespace from "${namespace}"`);
+      throw new MongoAPIError(`Cannot parse namespace from "${namespace}"`);
```

Comment:
this is runtime

Reply:
this will throw if namespace is [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy). I don't believe we currently pass in a user specified namespace directly to this function, but if we did this would be an API error. We have ticket where we want to roll the collection and Db's name validation into this class somehow.